components/test/src/main/java/org/trellisldp/test/ResourceServiceTests.java

```diff
@@ -88,7 +88,7 @@ default void testCreateResource() throws Exception {
         dataset.add(Trellis.PreferUserManaged, identifier, type, SKOS.Concept);
 
         assertFalse(getResourceService().get(identifier).isPresent());
-        assertTrue(getResourceService().create(identifier, getSession(), LDP.RDFSource, dataset).get());
+        assertTrue(getResourceService().create(identifier, getSession(), LDP.RDFSource, null, dataset).get());
```

Comment:
This `null` value is now going wreak havoc with the tests. Perhaps not with this particular test, but it will generally -- you can see the CI failures as an indication of that. I'd suggest replacing it with either: `getResourceService().getContainer(identifier)` or `final IRI container = rdf.createIRI(TRELLIS_DATA_PREFIX);`

The later option you could add as a class-wide field that gets re-used.

Reply:
On second thought, the first option will be the most flexible. Just replace `null` with `getResourceService().getContainer(identifier).orElse(null)`

Reply:
Wait, what? All tests pass for me. I believe you, I just want to understand why this is of concern; the only thing I understand the `null` to mean is that this isn't a contained resource-- to my thinking, that shouldn't fail any tests, unless they are specifically checking for containment. (Also, WTF do they pass for me?)

Reply:
I have no idea why the tests pass for you. They fail for me locally.

Specifically, in the `org.trellisldp.test.ResourceServiceTests` there are four tests: one for each of LDP-C, LDP-BC, LDP-DC and LDP-IC where containment triples are explicitly being counted. You can see an example [here](https://github.com/trellis-ldp/trellis/blob/master/components/test/src/main/java/org/trellisldp/test/ResourceServiceTests.java#L411).

In the `org.trellisldp.triplestore.TriplestoreResourceServiceTest` class, there is an expectation (internally) that an identified container will result in an extra `PreferServerManaged` triple. The fix here is either to leave the `null` value in the method and decrement the triple count in the test or leave the triple count the same and replace the `null` with a real `IRI`. In this [example](https://github.com/trellis-ldp/trellis/blob/ContainerParam/components/triplestore/src/test/java/org/trellisldp/triplestore/TriplestoreResourceServiceTest.java#L312-L325), the test is expecting 3 server-managed triples and a total of 5 triples. Keeping the `null` value in place, you'd want to change line 321 to `2L` and line 324 to `4L`.

Which is to say: the `null` value is just fine so long as the tests are not expecting to count the containment-related triples.

Reply:
Okay, that makes more sense-- counting containment triples is exactly what I _would_ expect to fail. But won't using `getResourceService().getContainer(identifier).orElse(null)` require more mocking (of that method in the `mockResourceService`)? Could I not just as well use a dummy IRI?

Reply:
e.g. `info:testContainer` or the like

Reply:
In this case, you shouldn't need any mocks. This test code is set up so that a _working_ `ResourceService` is used, and so while `info:testContainer` would probably work, it might be better for the actual `ResourceService` instance to manage its own parent-container definitions.

Reply:
Okay, I am now getting failures, so that's good (if really annoying). I will take a crack at the `getResourceService().getContainer(identifier).orElse(null)` approach.

Reply:
Nope, didn't fix the fails. Come to think of it, here's a common declaration for the id of resources in the tests:
```
 final IRI resource = rdf.createIRI(TRELLIS_DATA_PREFIX + "resource")
```
Now when stored via the java API, am I wrong in thinking that such a resource isn't going to get a "container" around it because it hasn't got one according to the path-munging algo? IOW, there is no path segment to interpret as a container.